### DISCUSSION TITLE
task are scheduled only after calling `vTaskStartScheduler()`

### DISCUSSION THREAD

**Discussion by kamomil** [feilipu/Arduino_FreeRTOS_Library#24] [Q&A]
Hi, I have arduino uno plus which is an arduino uno compatible.

It is commented:

>  // Now the Task scheduler, which takes over control of scheduling individual Tasks, is automatically started.

But I had to call `vTaskStartScheduler()`  in order for the tasks to run.

**Comment by feilipu** [comment]
The `vTaskStartScheduler()` is called during the [`initVariant()`](https://github.com/feilipu/Arduino_FreeRTOS_Library/blob/master/src/variantHooks.cpp#L92) call, which is the first thing that happens when the Arduino is booting up.

If you need to call `vTaskStartScheduler()` a second time, then you have other problems to address first, probably memory related.

**Comment by ubaldot** [comment]
I was reading the FreeRTOS and I get confused because in the examples here I cannot see `vTaskStartScheduler()`. But now I understand where it is "hidden" :)
